scripts/zkai_attention_kv_d64_four_head_seq32_air_private_softmax_table_lookup_gate.py

```
@@ -0,0 +1,919 @@
+
+
+def ensure_parent_dir_without_symlinks(path: pathlib.Path, label: str) -> None:
+    reject_symlinked_output_path(path, label)
+    candidate = path if path.is_absolute() else pathlib.Path.cwd() / path
+    current = pathlib.Path(candidate.anchor)
+    for part in candidate.parent.parts[1:]:
+        current = current / part
+        if current.is_symlink():
+            raise AttentionKvAirPrivateSoftmaxTableLookupGateError(
+                f"refusing to create {label} under symlinked parent: {current}"
+            )
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)